buildkite/resource_test_suite.go

```diff
@@ -0,0 +1,252 @@
+	return &testSuiteResource{}
+}
+
+func (ts *testSuiteResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	ts.client = req.ProviderData.(*Client)
+}
+
+func (ts *testSuiteResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state testSuiteModel
+	var response testSuiteResponse
+	payload := map[string]interface{}{}
+	var teamOwnerUuid string
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// The REST API requires team UUIDs but everything else in the provider uses GraphQL IDs. So we map from UUID to ID
+	// here
+	apiResponse, err := getNode(ts.client.genqlient, plan.TeamOwnerId.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to find team", err.Error())
+		return
+	}
+	if apiTeam, ok := apiResponse.Node.(*getNodeNodeTeam); ok {
+		teamOwnerUuid = apiTeam.Uuid
+	}
+
+	payload["name"] = plan.Name.ValueString()
+	payload["default_branch"] = plan.DefaultBranch.ValueString()
+	payload["show_api_token"] = true
+	payload["team_ids"] = []string{
+		teamOwnerUuid,
+	}
+
+	url := fmt.Sprintf("/v2/analytics/organizations/%s/suites", ts.client.organization)
+	err = ts.client.makeRequest("POST", url, payload, &response)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to create test suite", err.Error())
+		return
+	}
+
+	// TODO: dont rely on this, add the ID to the REST response
+	response.GraphqlID = base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("Suite---%s", response.UUID)))
+
+	state.ApiToken = types.StringValue(response.ApiToken)
+	state.DefaultBranch = types.StringValue(response.DefaultBranch)
+	state.ID = types.StringValue(response.GraphqlID)
+	state.UUID = types.StringValue(response.UUID)
+	state.Name = types.StringValue(response.Name)
+	state.Slug = types.StringValue(response.Slug)
+	state.TeamOwnerId = plan.TeamOwnerId
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+func (ts *testSuiteResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state testSuiteModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	url := fmt.Sprintf("/v2/analytics/organizations/%s/suites/%s", ts.client.organization, state.Slug.ValueString())
+	err := ts.client.makeRequest("DELETE", url, nil, nil)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to delete test suite", err.Error())
+		return
+	}
+}
+
+func (*testSuiteResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_test_suite"
+}
+
+func findTeamSuite(uuid string, suites []listTeamSuitesOrganizationSuitesSuiteConnectionEdgesSuiteEdge) *listTeamSuitesOrganizationSuitesSuiteConnectionEdgesSuiteEdgeNodeSuite {
```

Comment:
That's some name 😂

Reply:
No longer used actually so I'll remove it